packages/engine/lib/stateful/src/field/mod.rs

```diff
@@ -1,4 +1,11 @@
-//! TODO: DOC
+//! Provides an hEngine-specific representation of Arrows `Field` and `DataType`.
+//!
+//! As we are using only a subset of types provided by `Arrow`, our types are represented by
+//! [`FieldType`] rather than [`arrow::DataType`]. As [`FieldType`] can't be used in
+//! [`arrow::Field`], this module provides [`FieldSpec`] instead.
```

Comment:
I think this is still missing the point a little.

We provide a different **selection** of types to our users, which underneath uses a subset of `Arrow` types. It's not just that we present a subset, but we actively expose different types such as `Any` and `Number`. 

And I don't quite like the description of `FieldSpec`. I think we should say that fields in HASH simulations are defined through the `RootFieldSpec` and `FieldSpec` objects.

Reply:
> but we actively expose different types such as `Any` and `Number`

`Any` is just `String` with some requirements (JSON-Encoded) and `Number` is the same as a 64 bit floating point, so we actually only exposing a `String` type with some limitations. I updated that in the docs.

`FieldSpec` only associates a name and the `FieldType`, which provides the variant and `nullable`. Our `nullable` logic is a bit weird, because we couldn't use Arrow metadata before (this changed with the updated Arrow version) but we didn't overhaul it yet. This is why `nullable` is in `FieldType` instead of `FieldSpec`. Arrows `Field` strictly has the same data as `FieldSpec`.

I wouldn't mention `RootFieldSpec` here until we move that to a crate as well, which likely belongs to packages or at the same module as `FieldSpec`.

Reply:
I know what they are underneath, but that's not the point of our structs, which means this isn't a good explanation of the module. Reading the new description is, in my eyes, actually more confusing than previously and just drops in random terminology like `FieldSource` with little context.

> I wouldn't mention RootFieldSpec here until we move that to a crate as well, which likely belongs to packages or at the same module as FieldSpec.

Don't think I'd quite appreciated the fact that `RootFieldSpec` wasn't in this crate. Why is it not? I thought the reason we hadn't moved it was because of `FieldSource` but that's been changed.

Reply:
I'll add `RootFieldSpec` to the crate as well because, as you mentioned, it's no longer blocked on `FieldSource`.

Reply:
I added [`RootFieldSpec`] and related structs and overhauled the module-docs.